devTools/svgTester/verify-graphs.ts

```diff
@@ -0,0 +1,110 @@
+    const outDir = parsedArgs["o"] ?? "differentGrapherSvgs"
+    const numPartitions = parsedArgs["n"] ?? 1
+    const partition = parsedArgs["p"] ?? 1
+    const reverseDirectories = parsedArgs["l"] ?? false
+    const verbose = parsedArgs["v"] ?? false
+    // minimist turns a single number into a JS number so we do toString to normalize (TS types are misleading)
+    const rawGrapherIds: string = (parsedArgs["g"] ?? "").toString()
+
+    if (partition <= 0) throw "Partition must be >= 1"
+    if (partition > numPartitions) throw "Partition must be <= numPartitions"
+    if (numPartitions <= 0) throw "numPartitions must be >= 1"
+    if (numPartitions > 1000) throw "numPartitions must be <= 1000"
+    if (!fs.existsSync(inDir)) throw `Input directory does not exist ${inDir}`
+    if (!fs.existsSync(referenceDir))
+        throw `Reference directory does not exist ${inDir}`
+    if (!fs.existsSync(outDir)) fs.mkdirSync(outDir)
+
+    const grapherIds: number[] = utils.getGrapherIdListFromString(rawGrapherIds)
+    const directoriesToProcess = await utils.decideDirectoriesToVerify(
+        grapherIds,
+        inDir,
+        reverseDirectories,
+        numPartitions,
+        partition
+    )
+    const csvContentMap = await utils.getReferenceCsvContentMap(referenceDir)
+
+    const differences: number[] = []
+    for (const dir of directoriesToProcess) {
+        const [svg, svgRecord] = await utils.renderSvg(dir)
+
+        const referenceEntry = csvContentMap.get(svgRecord.chartId)
+        if (referenceEntry === undefined)
+            throw `Reference entry not found for ${svgRecord.chartId}`
+
+        const validationResult = await utils.verifySvg(
+            svg,
+            svgRecord,
+            referenceEntry,
+            referenceDir,
+            verbose
+        )
+
+        // verifySvg returns a Result type - if it is success we don't care any further
+        // but if there was an error then we write the svg and a message to stderr
+        switch (validationResult.kind) {
+            case "error":
+                utils.logDifferencesToConsole(svgRecord, validationResult)
+                const outputPath = path.join(outDir, svgRecord.svgFilename)
+                await fs.writeFile(outputPath, svg)
+                differences.push(svgRecord.chartId)
+        }
+    }
+
+    if (differences.length === 0) {
+        utils.logIfVerbose(
+            verbose,
+            `There were no differences in all ${directoriesToProcess.length} graphs processed`
+        )
+        process.exitCode = 0
+    } else {
+        console.warn(
+            `${
+                differences.length
+            } graphs had differences: ${differences.join()}`
+        )
+        for (const id of differences) {
+            console.log("", id) // write to stdout one grapher id per file for easy piping to other processes
+        }
+        process.exitCode = differences.length
+    }
```

Comment:
The function's looking better now, I would still suggest splitting lines 59-79 out into their own `displayResult` function, so that here it only looks like:

```
const ok = displayResults(validationResults)

process.exit(0 ? ok : 1)
```